rust/xaynet-analytics/src/controller.rs

```diff
@@ -84,50 +89,38 @@ impl<'ctrl> AnalyticsController<'ctrl> {
     }
 
     pub fn maybe_send_data(&mut self) -> Result<(), Error> {
-        if self.should_send_data() {
+        let can_send_data = self.is_charging && self.is_connected_to_wifi;
+        let should_send_data = can_send_data && !self.did_send_already_in_this_period();
+        if should_send_data {
             self.send_data()
         } else {
             Ok(())
         }
     }
 
-    fn check_for_new_screen_route(&self, event: &AnalyticsEvent) -> Result<(), Error> {
-        match event.screen_route {
-            Some(screen_route) => self.add_screen_route_if_new(screen_route),
-            None => Ok(()),
-        }
-    }
-
-    fn add_screen_route_if_new(&self, screen_route: &ScreenRoute) -> Result<(), Error> {
-        if !self
-            .screen_route_repo
-            .get_all(&self.db)?
-            .contains(screen_route)
+    fn add_screen_route_if_new(&self, screen_route_name: &str) -> Result<ScreenRoute, Error> {
+        let existing_screen_routes =
+            ScreenRoute::get_all(&self.db, &CollectionNames::SCREEN_ROUTES)?;
+        if let Some(existing_screen_route) = existing_screen_routes
+            .into_iter()
+            .find(|existing_route| existing_route.name == screen_route_name)
         {
-            self.screen_route_repo.add(screen_route, &self.db)
+            Ok(existing_screen_route)
         } else {
-            Ok(())
+            let screen_route = ScreenRoute::new(screen_route_name, Utc::now());
+            let screen_route_clone = screen_route.clone();
+            screen_route.add(&self.db, &CollectionNames::SCREEN_ROUTES)?;
+            Ok(screen_route_clone)
         }
     }
 
-    fn get_last_time_data_sent(
-        repo: &ControllerDataRepo,
-        db: &IsarDb,
-    ) -> Result<Option<DateTime<Utc>>, Error> {
-        match repo.get_all(db)?.last() {
+    fn get_last_time_data_sent(db: &IsarDb) -> Result<Option<DateTime<Utc>>, Error> {
+        match ControllerData::get_all(db, &CollectionNames::CONTROLLER_DATA)?.last() {
```

Comment:
This could be:
```rust
ControllerData::get_all(db, &CollectionNames::CONTROLLER_DATA)
.map(|data| data.last().map(|data| data.time_data_sent))
```
or
```rust
Ok(ControllerData::get_all(db, &CollectionNames::CONTROLLER_DATA)?
  .last()
  .map(|data| data.time_data_sent))
```

Reply:
will do